### ISSUE TITLE
`serde-sarif` doesn't deserialize additional properties

### ISSUE THREAD

**Issue by edoardopirovano** [psastras/sarif-rs#397]
Per [section 3.8](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317448) of the SARIF spec, a `properties` field can contain arbitrary extra data, i.e. fields with any name.  This is highlighted in the JSON schema that the package is built from [here](https://github.com/psastras/sarif-rs/blob/main/serde-sarif/src/schema.json#L1610). However, that fact seems to have not made it to the code and these extra properties are not currently deserialized by `serde-sarif`. It would be great if these could be deserialized, e.g. into a field like:
```rust
use serde_json::Value;
use std::collections::BTreeMap;
...
struct PropertyBag {
    ...
    #[serde(flatten)]
    additional_properties: BTreeMap<String, Value>,
}
```

**Comment by psastras** [comment]
@edoardopirovano thanks for reporting, this should be available in the latest published version of serde-sarif (0.3.5).

see example: https://github.com/psastras/sarif-rs/blob/main/serde-sarif/src/sarif.rs#L295

let me know if there is any issue, thanks!

**Comment by edoardopirovano** [comment]
Amazing, thanks for the quick turnaround on this!
